hazelcast-tpc-engine/src/main/java/com/hazelcast/internal/tpc/AsyncServerSocket.java

```diff
@@ -52,14 +53,12 @@ public long getAccepted() {
     public final SocketAddress getLocalAddress() {
         try {
             return getLocalAddress0();
-        } catch (Error e) {
-            throw e;
         } catch (Exception e) {
             return null;
         }
     }
 
-    protected abstract SocketAddress getLocalAddress0() throws Exception;
```

Comment:
Needs to be reverted to Exception. We have more to worry about than just NioSocket. There is also an IOUringSocket that can throw any form of exception.

Reply:
The goal of a 'throws Exception' is to say that the implementation can throw any kind of exception including any form of RuntimeException.